src/util.js

```
@@ -152,50 +147,19 @@ export async function deleteUsers (xo, userIds) {
 
 // ==================================================================
 
-export function getAllHosts (xo) {
-  return xo.objects.indexes.type.host
-}
-
-export function getOneHost (xo) {
-  const hosts = getAllHosts(xo)
-  for (const id in hosts) {
-    return hosts[id]
-  }
-
-  throw new Error('no hosts found')
-}
-
-// ==================================================================
-
-export async function getNetworkId (xo) {
-  const config = await getConfig()
-  const networks = xo.objects.indexes.type.network
-  const network = find(networks, {name_label: config.network})
-  return network.id
+export function getPoolTest () {
+  return find(xo.objects.all, {type: 'pool', name_label: config.pool})
 }
 
 // ==================================================================
 
-export async function getVmXoTestPvId (xo) {
-  const config = await getConfig()
-  const vms = xo.objects.indexes.type.VM
-  const vm = find(vms, {name_label: config.pvVm})
-  return vm.id
-}
-
-export async function getVmToMigrateId (xo) {
-  const config = await getConfig()
-  const vms = xo.objects.indexes.type.VM
-  const vm = find(vms, {name_label: config.vmToMigrate})
-  return vm.id
+export function getNetworkId () {
+  return find(xo.objects.all, {type: 'network', $poolId: getPoolTest().id}).id
 }
 
 // ==================================================================
-
-export async function getSrId (xo) {
-  const host = getOneHost(xo)
-  const pool = await xo.getOrWaitObject(host.$poolId)
-  return pool.default_SR
+export function getSrId () {
+  return getPoolTest().default_SR
```

Maybe you should fail if not defined?